cluster/watchdog.go

```diff
@@ -146,23 +154,6 @@ func (w *Watchdog) rescheduleContainers(e *Engine) {
 		// see https://github.com/docker/docker/issues/17750
 		// Add the global networks one by one
 		for networkName, endpoint := range globalNetworks {
-			hasSubnet := false
-			network := w.cluster.Networks().Uniq().Get(networkName)
-			if network != nil {
-				for _, config := range network.IPAM.Config {
-					if config.Subnet != "" {
-						hasSubnet = true
```

Comment:
The subnet is not mutated in either the previous version or the changes in this PR. What changed, is that in the previous version we would erase the IP address from the container network config IF the network does NOT have subnet specified, as the assumption is that fixed subnets would require fixed IPs. 

However, that condition is flawed for this code block here, as all networks have a subnet allocated at network creation time, and being able to allocate a specific IP address during container creation is not correlated to whether the swarm manager can detect that subnet or not.

Instead, this PR tweaks this logic so that "if the previous network disconnect failed, you cannot attempt to allocate the same IP address". This should be a more concrete condition, as there is a causal relationship between a disconnect operation for a given IP failing, and a network attachment request for that same IP failing in the future.